src/panels/AssetsFolderPanel/hooks.ts

```diff
@@ -0,0 +1,95 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { useOpenSpaceApi } from '@/api/hooks';
+
+import { Asset, Folder } from './types';
+import { baseName, pruneEmptyFolders } from './util';
+
+export function useAssetFolders() {
+  const luaApi = useOpenSpaceApi();
+  const [rootFolder, setRootFolder] = useState<Folder | null>(null);
```

Comment:
rootFolder is a little bit confusing here; maybe call it `folderStructure` or just `folders`?